Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PMLPsatdAlgorithmRZ.cpp

```diff
@@ -0,0 +1,176 @@
+/* Copyright 2021 David Grote
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#include "PMLPsatdAlgorithmRZ.H"
+#include "Utils/WarpXConst.H"
+#include "Utils/WarpXProfilerWrapper.H"
+#include "WarpX.H"
+
+#include <cmath>
+
+using amrex::operator""_rt;
+
+
+/* \brief Initialize coefficients for the update equation */
+PMLPsatdAlgorithmRZ::PMLPsatdAlgorithmRZ (SpectralKSpaceRZ const & spectral_kspace,
+                                          amrex::DistributionMapping const & dm,
+                                          const SpectralFieldIndex& spectral_index,
+                                          int const n_rz_azimuthal_modes, int const norder_z,
+                                          bool const nodal, amrex::Real const dt)
+     // Initialize members of base class
+     : SpectralBaseAlgorithmRZ(spectral_kspace, dm, spectral_index, norder_z, nodal),
+       m_spectral_index(spectral_index),
+       m_dt(dt)
+{
+    // Allocate the arrays of coefficients
+    amrex::BoxArray const & ba = spectral_kspace.spectralspace_ba;
+    C_coef = SpectralRealCoefficients(ba, dm, n_rz_azimuthal_modes, 0);
+    S_ck_coef = SpectralRealCoefficients(ba, dm, n_rz_azimuthal_modes, 0);
+
+    coefficients_initialized = false;
+}
+
+/* Advance the E and B field in spectral space (stored in `f`)
+ * over one time step */
+void
+PMLPsatdAlgorithmRZ::pushSpectralFields (SpectralFieldDataRZ & f)
+{
+
+    if (not coefficients_initialized) {
+        // This is called from here since it needs the kr values
+        // which can be obtained from the SpectralFieldDataRZ
+        InitializeSpectralCoefficients(f);
+        coefficients_initialized = true;
+    }
+
+    const SpectralFieldIndex& Idx = m_spectral_index;
+
+    // Loop over boxes
+    for (amrex::MFIter mfi(f.fields); mfi.isValid(); ++mfi){
+
+        amrex::Box const & bx = f.fields[mfi].box();
+
+        // Extract arrays for the fields to be updated
+        amrex::Array4<Complex> const& fields = f.fields[mfi].array();
+        // Extract arrays for the coefficients
+        amrex::Array4<const amrex::Real> const& C_arr = C_coef[mfi].array();
+        amrex::Array4<const amrex::Real> const& S_ck_arr = S_ck_coef[mfi].array();
+
+        // Extract pointers for the k vectors
+        auto const & kr_modes = f.getKrArray(mfi);
```

Comment:
This could work like this, but better to double check:
```suggestion
        RealVector const & kr_modes = f.getKrArray(mfi);
```

Reply:
I used the full type `HankelTransform::RealVector`.